components/org.wso2.config.diff.creater/src/main/java/org/wso2/config/diff/creater/ConfigDiffChecker.java

```diff
@@ -0,0 +1,339 @@
+                        Paths.get(outputCSVFile.getPath()), Paths.get(keyValueFile.getPath()), changedPropertyDiffSet);
+            } else {
+
+                log.warn(entry.getValue().getPath() + " is not templated with toml. \n");
+                File outFile =
+                        new File(MigrationConstants.UN_TEMPLATE_FILE_FOLDER + MigrationConstants
+                                .FILE_SEPARATOR + entry.getKey());
+                FileUtils.copyFile(entry.getValue(), outFile);
+            }
+        }
+    }
+
+
+    /**
+     * Find the difference of property files.
+     *
+     * @param defaultFile  default file to compare with.
+     * @param migratedFile migrated file used to compare.
+     * @return A set of difference.
+     * @throws IOException IOException.
+     */
+    private Set<Map.Entry<String, String>> findDiffPropertiesFiles(File defaultFile, File migratedFile)
+            throws IOException {
+
+        Properties defaultProperties = new Properties();
+        Properties changedProperties = new Properties();
+        try (FileInputStream defaultInputStream = new FileInputStream(defaultFile);
+             FileInputStream changedInputStream = new FileInputStream(migratedFile)) {
+
+            defaultProperties.load(defaultInputStream);
+            changedProperties.load(changedInputStream);
+            Map<String, String> defaultMap = new HashMap<String, String>((Map) defaultProperties);
+            Map<String, String> changedMap = new HashMap<String, String>((Map) changedProperties);
+
+            Set<Map.Entry<String, String>> defaultPropertySet = defaultMap.entrySet();
+            Set<Map.Entry<String, String>> changedPropertySet = changedMap.entrySet();
+            // Leaves only entries in changedPropertySet that are only in changedPropertySet, not in defaultPropertySet.
+            changedPropertySet.removeAll(defaultPropertySet);
+
+            return changedPropertySet;
+        } catch (IOException e) {
+            throw new IOException("The configurations can not be loaded properly. Please try again. ");
+        }
+    }
+
+    /**
+     * Write difference to csv files.
+     *
+     * @param migratedFile       migrated file.
+     * @param existingTags       existing tags in the knowledge base.
+     * @param outputCSVFilePath  output csv file path.
+     * @param keyValueFilePath   key-value csv file path.
+     * @param changedPropertySet property difference.
+     * @throws IOException IOException.
+     */
+    private void writePropertiesDiffToCSV(File migratedFile, Map<String, String> existingTags, Path outputCSVFilePath,
+                                          Path keyValueFilePath, Set<Map.Entry<String, String>> changedPropertySet)
+            throws IOException {
+
+        for (Map.Entry<String, String> property : changedPropertySet) {
+
+            String keyValueData;
+            String csvEntry;
+
+            if (existingTags.get(property.getKey()) == null) {
```

Comment:
By any chance can existingTags null here?

Reply:
no because the csv file that we have hosted  is anyway with values